opt/api/utils/http/HttpStringBuilder.java

```diff
@@ -0,0 +1,49 @@
+package opt.api.utils.http;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class HttpStringBuilder {
+    
+    public static String build(BufferedReader in) {
+        StringBuilder requestBuilder = new StringBuilder();
+        String line;
+
+        try {
+            while ((line = in.readLine()) != null && !line.trim().isEmpty()) {
```

Comment:
The entire request looks like:
```
POST HTTP/1.1 /route
Header-One: Value
Header-Two: Value

{"body": "value"}
```

There's a blank line above the body (which is what I want to read) but `in.readLine()` takes in the next line, which is blank and then sits waiting for input above the body. There doesn't seem to be a way of making it move onto the next line and reading all the body

Reply:
I've pull this and have a look!

Reply:
It doesn't hang, the while completes at the blank line.
Remove the second condition in the while: line/trim/isEmpty.
A main program to test this function:

    public static void main(String[] args) {
    	String request = 
    			"POST HTTP/1.1 /route\n"
    			+"Header-One: Value\n"
    			+"Header-Two: Value\n"
    			+"\n"
    			+"{'body': 'value'}";
    	Reader inputString = new StringReader( request );
    	BufferedReader br = new BufferedReader( inputString );
    	String output = build( br );
    	System.out.println( "request is\n"+ output );
    }